soroban-env-host/src/host.rs

```diff
@@ -847,31 +910,34 @@ impl EnvBase for Host {
 impl CheckedEnv for Host {
     type Error = HostError;
 
+    // Notes on metering: covered by the components
     fn log_value(&self, v: RawVal) -> Result<RawVal, HostError> {
         self.record_debug_event(DebugEvent::new().msg("log").arg(v))?;
         Ok(RawVal::from_void())
     }
 
+    // Notes on metering: covered by the components
     fn get_invoking_contract(&self) -> Result<Object, HostError> {
         let frames = self.0.context.borrow();
         // the previous frame must exist and must be a contract
         let hash: Hash = if frames.len() >= 2 {
             match &frames[frames.len() - 2] {
                 #[cfg(feature = "vm")]
-                Frame::ContractVM(vm) => Ok(vm.contract_id.clone()),
+                Frame::ContractVM(vm) => Ok(vm.contract_id.metered_clone(&self.0.budget)?),
                 Frame::HostFunction(_) => {
                     Err(self.err_general("Host function context has no contract ID"))
                 }
                 Frame::Token(id) => Ok(id.clone()),
                 #[cfg(feature = "testutils")]
-                Frame::TestContract(id) => Ok(id.clone()),
+                Frame::TestContract(id) => Ok(id.metered_clone(&self.0.budget)?),
             }
         } else {
             Err(self.err_general("no invoking contract"))
         }?;
         Ok(self.add_host_object(<Vec<u8>>::from(hash.0))?.into())
     }
 
+    // Notes on metering: covered by `unchecked_visit_val_obj`.
```

Comment:
You are absolutely right. Unfortunately this will take some extra work, which I've created a new tracking issue for it: https://github.com/stellar/rs-soroban-env/issues/373. I will probably do something similar as `MeteredClone` in here (i.e. `MeteredCmp`), and impl them for every object type.